tests/src/test/java/org/eclipse/hono/tests/registry/CredentialsAmqpIT.java

```diff
@@ -119,37 +134,55 @@ public void testGetCredentialsNotExistingAuthId(final TestContext ctx) {
 
     /**
      * Verifies that the service returns credentials for a given type and authentication ID.
-     * 
+     *
      * @param ctx The vert.x test context.
      */
     @Test
     public void testGetCredentialsReturnsCredentialsTypeAndAuthId(final TestContext ctx) {
 
-        credentialsClient
-            .get(CredentialsConstants.SECRETS_TYPE_HASHED_PASSWORD, CREDENTIALS_AUTHID1)
-            .setHandler(ctx.asyncAssertSuccess(result -> {
-                ctx.assertEquals(CREDENTIALS_AUTHID1, result.getAuthId());
-                ctx.assertEquals(CredentialsConstants.SECRETS_TYPE_HASHED_PASSWORD, result.getType());
-            }));
+        final String deviceId = helper.getRandomDeviceId(Constants.DEFAULT_TENANT);
+        final CredentialsObject payload = new CredentialsObject(deviceId, CREDENTIALS_AUTHID1, CredentialsConstants.SECRETS_TYPE_HASHED_PASSWORD);
+        getAuthId1Secrets().forEach( secret -> payload.addSecret((JsonObject) secret));
+
+        helper.registry.addCredentials(Constants.DEFAULT_TENANT, JsonObject.mapFrom(payload))
+                .compose(r -> credentialsClient.get(CredentialsConstants.SECRETS_TYPE_HASHED_PASSWORD, CREDENTIALS_AUTHID1))
+                .compose(result -> {
+                    ctx.assertEquals(CREDENTIALS_AUTHID1, result.getAuthId());
+                    ctx.assertEquals(CredentialsConstants.SECRETS_TYPE_HASHED_PASSWORD, result.getType());
+                    return Future.succeededFuture();
+                }).setHandler(ctx.asyncAssertSuccess());
     }
 
-    /**
-     * Verifies that the service returns credentials for a given type, authentication ID and matching client context.
-     *
-     * @param ctx The vert.x test context.
-     */
+/**
+ * Verifies that the service returns credentials for a given type, authentication ID and matching client context.
+ *
+ * @param ctx The vert.x test context.
+ */
     @Test
     public void testGetCredentialsExistingClientContext(final TestContext ctx) {
 
+        final String deviceId = helper.getRandomDeviceId(Constants.DEFAULT_TENANT);
+        final CredentialsObject payload = new CredentialsObject(deviceId, "gw", CredentialsConstants.SECRETS_TYPE_HASHED_PASSWORD);
+
+        payload.addSecret( new JsonObject()
```

Comment:
simply use a corresponding factory method, e.g. `CredentialsObject.fromClearTextPassword()` which saves a lot of boiler plate code ...